paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentSheetViewModelTest.kt

```diff
@@ -828,26 +834,74 @@ internal class PaymentSheetViewModelTest {
             )
     }
 
-    fun `Verify supported payment methods exclude afterpay if no shipping`() {
+    @Test
+    fun `Verify supported payment methods exclude afterpay if no shipping and no allow flag`() {
+        val viewModel = createViewModel(
+            args = ARGS_CUSTOMER_WITH_GOOGLEPAY.copy(
+                config = ARGS_CUSTOMER_WITH_GOOGLEPAY.config?.copy(
+                    shippingDetails = null,
+                    allowsPaymentMethodsRequiringShippingAddress = false,
+                )
+            )
+        )
+
         viewModel.setStripeIntent(
             PaymentIntentFixtures.PI_WITH_SHIPPING.copy(
                 paymentMethodTypes = listOf("afterpay_clearpay"),
-                shipping = null
+                shipping = null,
             )
         )
 
-        assertThat(viewModel.supportedPaymentMethods.size).isEqualTo(0)
+        assertThat(viewModel.supportedPaymentMethods).isEmpty()
+    }
+
+    @Test
+    fun `Verify supported payment methods include afterpay if allow flag but no shipping`() {
+        val viewModel = createViewModel(
+            args = ARGS_CUSTOMER_WITH_GOOGLEPAY.copy(
+                config = ARGS_CUSTOMER_WITH_GOOGLEPAY.config?.copy(
+                    shippingDetails = null,
+                    allowsPaymentMethodsRequiringShippingAddress = true,
+                )
+            )
+        )
 
         viewModel.setStripeIntent(
             PaymentIntentFixtures.PI_WITH_SHIPPING.copy(
```

Comment:
Just checking, should this set `shipping = null` since this test says "no shipping"?

Reply:
Right! I was confused between the config’s shipping details and the intent’s shipping details.